### ISSUE TITLE
Narrator/NVDA is not announcing as selected for the selected date.

### ISSUE THREAD

**Issue by msftedad** [Hacker0x01/react-datepicker#4671]
**Describe the bug**
Narrator/NVDA is not announcing as selected for the selected date.

**To Reproduce**

1. Hit the URL "https://jqueryui.com/datepicker/".
2. Tab till Date edit box and hit Enter key.
3. Tab till selected date.
4. Observe the issue

**Expected behavior**
Narrator/NVDA should announce as selected for the selected date and it should announce as current for the current(todays) date.

**Screenshots**
[a168985f-13fc-4c9a-bbc8-507d0f0afb37.webm](https://github.com/Hacker0x01/react-datepicker/assets/93735775/3a7a0afc-da32-440d-a79c-d6af8e4368ba)
[32f18396-3608-4117-8112-6d888b02384c.webm](https://github.com/Hacker0x01/react-datepicker/assets/93735775/2e7c0b08-bc3f-4384-8d4f-833c3d1f94b8)

**Desktop (please complete the following information):**

- OS: Windows 11
- Browser: Edge, Chrome
- Version: 122.0.2365.92 (Official build) (64-bit), 

**Smartphone (please complete the following information):**

**Additional context**

**Comment by msftedad** [comment]
Please dont close this issue as this is still repro'ing.
